File: webstore/webstore-java/src/main/java/com/renan/webstore/persistence/repository/UserRepository.java
package com.renan.webstore.persistence.repository;

import com.renan.webstore.model.AppUser;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<AppUser, Integer> {
    
    Optional<AppUser> findByEmail(String email);
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/persistence/repository/UserRepository.kt
package com.renan.webstore.persistence.repository

import com.renan.webstore.model.AppUser
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserRepository: JpaRepository<AppUser, Int> {

    fun findByEmail(email: String): Optional<AppUser>

}